### ISSUE TITLE
TypeError: Cannot read property 'properties' of undefined

### ISSUE THREAD

**Issue by azranel** [netguru/eslint-plugin-netguru-ember#27]
I just had this problem

```
File: app.js
The Broccoli Plugin: [EslintValidationFilter] failed with:
TypeError: Cannot read property 'properties' of undefined
    at Object.getModuleProperties (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/eslint-plugin-netguru-ember/rules/utils/ember.js:270:66)
    at EventEmitter.CallExpression (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/eslint-plugin-netguru-ember/rules/order-in-routes.js:21:30)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
    at NodeEventGenerator.enterNode (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/eslint/lib/util/node-event-generator.js:40:22)
    at CodePathAnalyzer.enterNode (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at CommentEventGenerator.enterNode (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/eslint/lib/util/comment-event-generator.js:97:23)
    at Controller.enter (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/eslint/lib/eslint.js:925:36)
    at Controller.__execute (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/eslint/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/eslint/node_modules/estraverse/estraverse.js:501:28)

The broccoli plugin was instantiated at:
    at EslintValidationFilter.Plugin (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/broccoli-plugin/index.js:7:31)
    at EslintValidationFilter.Filter [as constructor] (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/broccoli-persistent-filter/index.js:60:10)
    at new EslintValidationFilter (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/broccoli-lint-eslint/build/index.js:111:10)
    at EslintValidationFilter (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/broccoli-lint-eslint/build/index.js:94:12)
    at Class.module.exports.lintTree (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/ember-cli-eslint/index.js:31:12)
    at /Users/bartoszlecki/netguru/ForTheOne-front/node_modules/ember-cli/lib/broccoli/ember-app.js:603:20
    at Array.map (native)
    at EmberApp.addonLintTree (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/ember-cli/lib/broccoli/ember-app.js:601:36)
    at EmberApp.lintTestTrees (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/ember-cli/lib/broccoli/ember-app.js:1158:24)
    at EmberApp.appTests (/Users/bartoszlecki/netguru/ForTheOne-front/node_modules/ember-cli/lib/broccoli/ember-app.js:1129:51)
```

From what I saw https://github.com/netguru/eslint-plugin-netguru-ember/blob/master/rules/utils/ember.js#L270 there is no undefined check.

**Comment by azranel** [comment]
More specifically such code is not accepted by this plugin

```
import AjaxErrorMixin from './app/mixins/ajax-error';

Ember.Route.reopen(AjaxErrorMixin);
Ember.Controller.reopen(AjaxErrorMixin);
```

From what I see when you don't provide `{}` to reopen method it brakes 😄

**Comment by azranel** [comment]
This fixes issue:
```
Ember.Route.reopen(AjaxErrorMixin, {});
Ember.Controller.reopen(AjaxErrorMixin, {});
```

Nonetheless I think that first form should be valid as well (actually it's less redundant).

**Comment by caseywatts** [comment]
I just had the same sort of error, thanks for posting this @azranel ! :)

```
node_modules/.bin/eslint app/components/app/settings/app-name.js
Cannot read property 'properties' of undefined
TypeError: Cannot read property 'properties' of undefined
    at Object.getModuleProperties (/Users/cwatts/Code/dashboard/node_modules/eslint-plugin-netguru-ember/rules/utils/ember.js:270:66)
    at EventEmitter.CallExpression (/Users/cwatts/Code/dashboard/node_modules/eslint-plugin-netguru-ember/rules/no-on-calls-in-components.js:35:41)
    at emitOne (events.js:101:20)
    at EventEmitter.emit (events.js:188:7)
    at NodeEventGenerator.enterNode (/Users/cwatts/Code/dashboard/node_modules/eslint/lib/util/node-event-generator.js:40:22)
    at CodePathAnalyzer.enterNode (/Users/cwatts/Code/dashboard/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at CommentEventGenerator.enterNode (/Users/cwatts/Code/dashboard/node_modules/eslint/lib/util/comment-event-generator.js:97:23)
    at Controller.enter (/Users/cwatts/Code/dashboard/node_modules/eslint/lib/eslint.js:925:36)
    at Controller.__execute (/Users/cwatts/Code/dashboard/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (/Users/cwatts/Code/dashboard/node_modules/estraverse/estraverse.js:501:28)
```

That file has this line as the culprit; same idea:
```
export default Ember.Component.extend(EditableControllerMixin, RenameablePurpleInlineMixin);
```

**Comment by caseywatts** [comment]
Seems like this line could be made to check whether the `node` exists before trying to access it:

https://github.com/netguru/eslint-plugin-netguru-ember/blob/master/rules/utils/ember.js#L269

**Comment by caseywatts** [comment]
Made a PR that should fix this behavior 🎉 
https://github.com/netguru/eslint-plugin-netguru-ember/pull/29

Thanks again @azranel :)

**Comment by michalsnik** [comment]
Thank you guys. Fix has been shipped @ `1.6.1` 🎉
